students/Gorthi_kavita/10module/10assignement.py

```diff
@@ -0,0 +1,173 @@
+class donar():
+    list =[]   #this is donar list adding  to the list
+    allist =[]   #to keep track how many times the donar name in the list
+    uniquename = []  # unique donar names
+    listsum = []     # donar names and total amount and number of times donated
+    listadd =[]
+    finalsum = []
+
+    def __data(self,name,amount):
+        self.donarname = name
+        self.donaramount = amount
+        c = (name,amount,1)
+        d = (name)
+        self.list.append(c)
+        self.allist.append(d)
+
+    def donarname(self):
+        self.__data("mike",8)
+        self.__data("ketty",75)
+        self.__data("john hoop",45)
+        self.__data("ravi",10)
+        self.__data("mike",6)
+        self.__data("joy",55)
+        self.__data("david",12)
+        self.__data("david",4)
+
+
+    def  countname(self):
+        import collections
+        k = self.allist
+        counter = collections.Counter(k)       #how many times the donar name in the donation list
+        uniname = set(self.allist)
+        for i in uniname:
+            self.uniquename.append(i)          #this is unique names of the donar
+
+        for i in self.uniquename:        #this will find the sum of the donation amount and number of times donated
+            sum = 0
+            times = 0
+            for j in self.list:
+                 if(i == j[0]):
+                    sum = sum+j[1]
+                    times = times+j[2]
+                    c =(j[0],sum,times)
+                    if(counter[i] == times):
+                      self.listsum.append(c)
+
+    def  report(self):
+         print("+----------------------------+------------------+-------------------+-------------------+")
+         print("|     Donar Name             |  Total given     | Number of gifts   |    Average gift   |")
+         print("+----------------------------+------------------+-------------------+--------------------")
+
+         for i in self.listsum:
+             name = i[0].upper()
+             total = '$' + str(i[1])
+             avg = i[2]
+             k  = int(i[1]/i[2])
+             avggift = '$' + str(k)
+             print("{: >20} {: >20} {: >20} {: >20}".format(name, total, avg,avggift))
+
+    def  sendmail(self,listm):
+        listd = listm
+        for i in listd:
+            a = i[0].upper()
+            b = '$' + str(i[1])
+            file = open("{}"".txt".format(a), 'a+')
+            file.write("  Dear  ""{}".format(a))
+            file.write("\n")
+            file.write(" \n")
+            file.write("    Thank you for your very kind donation of ")
+            file.write("{}".format(b))
+            file.write("\n")
+            file.write("\n")
+            file.write("    It will be used for very good cause.")
+            file.write("\n")
+            file.write("           Sincerely, \n")
+            file.write("          -The Team  ")
+            file.close
```

Comment:
You could create this whole string first before you do your `file.write()`.  That way you aren't calling that method over and over again, just once.  Also :) `file` is a built in as well, need to call this something else.  Pycharm will warn you of these built ins.